diff --git a/dist/changesets-apply-release-plan.cjs.js b/dist/changesets-apply-release-plan.cjs.js
index e71b99fa922361c1d91d591ec15b4ed9c3689779..806942b22f59130b01cf7d5957a118532069dcba 100644
--- a/dist/changesets-apply-release-plan.cjs.js
+++ b/dist/changesets-apply-release-plan.cjs.js
@@ -208,5 +208,8 @@ async function getChangelogEntry(cwd, release, releases, changesets, changelogFu
   });
   let relevantChangesets = changesets.filter(cs => relevantChangesetIds.has(cs.id));
   changelogLines.patch.push(changelogFuncs.getDependencyReleaseLine(relevantChangesets, dependentReleases, changelogOpts));
+  if (typeof changelogFuncs.getChangelogEntry === "function") {
+    return changelogFuncs.getChangelogEntry(release, changelogLines);
+  }
   return [`## ${release.newVersion}`, await generateChangesForVersionTypeMarkdown(changelogLines, "major"), await generateChangesForVersionTypeMarkdown(changelogLines, "minor"), await generateChangesForVersionTypeMarkdown(changelogLines, "patch")].filter(line => line).join("\n");
 }
