diff --git a/dist/esm/index.mjs b/dist/esm/index.mjs
index e065926b3da86406660aba987cb8226af9f0f8fd..ee9e2bda075353d919e689a3077f22ec6ca9cb5c 100644
--- a/dist/esm/index.mjs
+++ b/dist/esm/index.mjs
@@ -67,21 +67,21 @@ function solidPlugin(options = {}) {
 
       // fix for bundling dev in production
       const nestedDeps = replaceDev ? ['solid-js', 'solid-js/web', 'solid-js/store', 'solid-js/html', 'solid-js/h'] : [];
-      const test = userConfig.test || {};
+      // TODO: undo patch once this is released - https://github.com/solidjs/vite-plugin-solid/pull/183
+      const userTest = userConfig.test ?? {};
+      const test = {};
       if (userConfig.mode === 'test') {
         // to simplify the processing of the config, we normalize the setupFiles to an array
-        const userSetupFiles = typeof test.setupFiles === 'string' ? [test.setupFiles] : test.setupFiles || [];
-        if (!test.environment && !options.ssr) {
+        const userSetupFiles = typeof userTest.setupFiles === 'string' ? [userTest.setupFiles] : userTest.setupFiles || [];
+        if (!userTest.environment && !options.ssr) {
           test.environment = 'jsdom';
         }
-        test.server = test.server || {};
-        test.server.deps = test.server.deps || {};
-        if (!test.server.deps.external?.find(item => /solid-js/.test(item.toString()))) {
-          test.server.deps.external = [...(test.server.deps.external || []), /solid-js/];
+        if (!userTest.server?.deps?.external?.find(item => /solid-js/.test(item.toString()))) {
+          test.server = {deps: {external: [ /solid-js/]}}
         }
         const jestDomImport = getJestDomExport(userSetupFiles);
         if (jestDomImport) {
-          test.setupFiles = [...userSetupFiles, jestDomImport];
+          test.setupFiles = [jestDomImport];
         }
       }
       return {
